ursh/indico_ursh/controllers.py

```diff
@@ -41,25 +42,33 @@ def _resolve_full_url(original_url):
     @staticmethod
     def _request_short_url(original_url):
         from indico_ursh.plugin import UrshPlugin
+        api_key = UrshPlugin.settings.get('api_key')
+        api_host = UrshPlugin.settings.get('api_host')
+        if not api_key or not api_host:
+            raise BadRequest(_('API key/host missing - have you set the plugin settings?'))
         headers = {'Authorization': 'Bearer {api_key}'.format(api_key=UrshPlugin.settings.get('api_key'))}
         response = requests.post(UrshPlugin.settings.get('api_host'),
                                  data=dict(url=original_url, allow_reuse=True),
                                  headers=headers).json()
         return posixpath.join(response['host'], response['shortcut'])
 
+    @staticmethod
+    def _check_host(full_url):
+        if url_parse(full_url).host != url_parse(config.BASE_URL).host:
+            raise BadRequest(_('Invalid host for URL shortening service'))
+
     def _process(self):
         try:
             original_url = request.json.get(u'original_url')
             full_url = self._resolve_full_url(original_url)
-            if url_parse(full_url).host != url_parse(config.BASE_URL).host:
-                raise BadRequest('Invalid host for URL shortening service')
+            self._check_host(full_url)
             short_url = self._request_short_url(full_url)
             return jsonify(success=True, msg=short_url)
         except BadRequest as e:
             return jsonify(success=False, msg=e.description)
         except RequestException:
-            return jsonify(success=False, msg='Failed to connect to URL shortening service')
+            return jsonify(success=False, msg=_('Failed to connect to URL shortening service'))
```

Comment:
I would just let it fail - that way users get the error report dialog and can send a report if they care enough. Also, the error will go to sentry in that case (if enabled)